### ISSUE TITLE
The 4 Errors of Dandere2x 3.6

### ISSUE THREAD

**Issue by DCGYT** [akai-katto/dandere2x#259]
Here are the 4 bugs of Dandere2x, if fixed it would be the perfect program for upscaling.

![image](https://user-images.githubusercontent.com/98280315/188295008-d9c50ee2-64ad-4e11-84f5-1087b574c709.png)
![image](https://user-images.githubusercontent.com/98280315/188295525-3d2a46a5-240b-4e74-b882-fb3aace6b4af.png)
![image](https://user-images.githubusercontent.com/98280315/188294968-8302f65c-ce47-4861-ab9c-0597bedb9b9c.png)
![image](https://user-images.githubusercontent.com/98280315/188295034-45f1d656-80ec-4deb-9ad0-8dccb8cd1e8f.png)

in this video I show you everything:
https://drive.google.com/file/d/14-mQ0v3KjpqDnRjJnWxWcSqS0ejWED6v/view?usp=sharing

No More Than 13 minutes File: https://drive.google.com/file/d/1vwDR-DnOMYPoITBcFU4WhmS09pQ-OMWd/view?usp=sharing
No More Than 13 minutes Log:https://drive.google.com/file/d/1rNagrkSacb5w9e271kbFcjWWMbyFdLrK/view?usp=sharing
White Dots File: https://drive.google.com/file/d/1gOJWDWAX45s8wSDr59Gc-qBWOdbL3gmE/view?usp=sharing

**Comment by akai-katto** [comment]
This is useful feedback to me as a developer. I'll set myself as the assignee.

**Comment by eoiz** [comment]
I upscaled the file but the result seems fine to me...maybe it's a player issue?
For settings I used Image Quality 100 and Block Size 60 with h265 encoding.
For the record: MediaInfo does detects 1 error (for me this is really common, I usually ALWAYS remux an episodie for audio to works correctly)
![image](https://user-images.githubusercontent.com/86598637/188616593-5838e592-4b83-4671-8527-e828458dca3a.png)

Video https://drive.google.com/file/d/1AgxSbgctszlZR3z50KbXE_u37--cg8Qf/view?usp=sharing
Log https://drive.google.com/file/d/1WHud1iPpjPH-gp7joTvPnOeKPxI4XnjE/view?usp=sharing

**Comment by DCGYT** [comment]
@eoiz Maybe it's because I rendered in x264 or that I'm on Windows 11.

I could try rendering in h265 but I don't know how to do it. It's just modifying the "output_options.yaml" in "vcodec" right? What text should I replace it with?

**Comment by eoiz** [comment]
@DCGYT I'm really sorry, on the test before I used the 3.5 version with my personal configuration, my bad...
Now I've re-upscaled it with version 3.6 and standard configuration (h264 100 quality 60 block), but it works fine for me...
Try download my version and check if it still blocks after 13 minutes.

Video https://drive.google.com/file/d/1EZToCt-cOvJlaFQouUJCK6-X3BTRlsMi/view?usp=sharing
Log https://drive.google.com/file/d/10y-pFKifBfrFDAvgsVQy8fbwVj0tPilL/view?usp=sharing

**Comment by eoiz** [comment]
Also @akai-katto I've noticed that the 3.6 version is slower overall. Trying to upscale (everything on default settings) the example movie in the dandere2x folder: 3.6 version (17,1 MB output) takes 3 minutes and the 3.5 version (16,6 MB output) took less than 2.

**Comment by akai-katto** [comment]
Are you sure? The console output is how long the application has been active, not how long the upscale took.

**Comment by akai-katto** [comment]
For what its worth, I can't replicate the 13 minute limitation. What's your computer specs?

**Comment by eoiz** [comment]
> Are you sure? The console output is how long the application has been active, not how long the upscale took.

Yes, it seems the it skips a step, but process the images slowly...
I also tried upscaling the video from DCGYT and the results are also confirming that:
version 3.5 took 45 minutes https://drive.google.com/file/d/13yskgdrU4iu3yD0T4AAlPZp0c1y5qxbi/view?usp=sharing
and version 3.6 took 78 minutes https://drive.google.com/file/d/1f91v5ucg9AIFQj15pY3KxsUTFmbWhBzN/view?usp=sharing

**Comment by DCGYT** [comment]
@eoiz It is quite strange, I have tried several ways and when I start the log it tells me directly 15314 frames. And you get 30755 (This did not happen in version 3.5).

**Comment by DCGYT** [comment]
Sorry my mistake, I made a mistake in the file, I will compare again.

**Comment by DCGYT** [comment]
@akai-katto Every time I try to upscale the file "NoMoreThan13Minutes.mkv" I get a different number of frames.
![image](https://user-images.githubusercontent.com/98280315/189014757-8140b59c-1453-48ee-a400-bd7e4a3214db.png)
![image](https://user-images.githubusercontent.com/98280315/189014924-c7bf303f-c8f8-450c-b0ae-5d928d41c276.png)
![image](https://user-images.githubusercontent.com/98280315/189014982-f568ac12-b178-4bf1-b486-a42e4515b81c.png)
![image](https://user-images.githubusercontent.com/98280315/189015197-860f51c8-8a93-4da5-a736-a88c875438cd.png)

**Comment by akai-katto** [comment]
Whats your computer specs?

**Comment by akai-katto** [comment]
Does using this binary help with anything? 

https://drive.google.com/file/d/15tQKEGyfNaFgRUKVvJxq309YoSYuAGek/view?usp=sharing

**Comment by DCGYT** [comment]
@akai-katto 
Intel Core i5 8300H 12GB RAM
Nvidia GeForce GTX 1050 Mobile 4GB
SSD Western Digital SN750 1TB
Windows 11 Home Single Language (Spanish-MX)

With the new "main" it only upscaled me 30 seconds.
workspace files: https://drive.google.com/drive/folders/1RbvhgSmUc1IXpeGojHHGyTqYG2vChEam?usp=sharing

**Comment by akai-katto** [comment]
So some things addressed:

- the reason why it's slower is because the quality is higher. I found this out when comparing the sizes of the input files to dandere2x. Try dropping the quality by 2-3 values to get the same speeds as before, thats why I had to do.
- I'm still looking into the 13 minute issue for me. Thanks for trying out the new binary

**Comment by akai-katto** [comment]
https://drive.google.com/file/d/1E22ENXO9srJAqFUAL9YfVo9tpzPS6aJi/view?usp=sharing

expect this to run slower, but does it give consistent frame counts?

**Comment by DCGYT** [comment]
@akai-katto The latest binary is giving me the same results as the original 3.6 version. Only the GUI has white letters.
![image](https://user-images.githubusercontent.com/98280315/189368166-80ea7d64-cffb-47e9-927f-6129d6a120e1.png)
![image](https://user-images.githubusercontent.com/98280315/189368290-465f0c43-700b-4136-82e5-903129fd95a8.png)
![image](https://user-images.githubusercontent.com/98280315/189368847-1162877d-6ab9-4490-b87f-57beeee5eb01.png)

**Comment by eoiz** [comment]
> * the reason why it's slower is because the quality is higher. I found this out when comparing the sizes of the input files to dandere2x. Try dropping the quality by 2-3 values to get the same speeds as before, thats why I had to do.

In the 3.5 vs 3.6 test I did I set the quality as default (98)...or wich quality do you intend?

**Comment by akai-katto** [comment]
> @akai-katto The latest binary is giving me the same results as the original 3.6 version. Only the GUI has white letters. ![image](https://user-images.githubusercontent.com/98280315/189368166-80ea7d64-cffb-47e9-927f-6129d6a120e1.png) ![image](https://user-images.githubusercontent.com/98280315/189368290-465f0c43-700b-4136-82e5-903129fd95a8.png) ![image](https://user-images.githubusercontent.com/98280315/189368847-1162877d-6ab9-4490-b87f-57beeee5eb01.png)

It prints out 
```
root 2022-09-12 21:38:46,509 INFO     ffprobe.py get_video_info: Command: ['C:\\Users\\windw0z\\Documents\\GitHub\\dandere2x\\src\\externals\\ffprobe.exe', '-v', 'panic', '-print_format', 'json', '-show_format', '-show_streams', '-i', 'C:\\Users\\windw0z\\Documents\\GitHub\\dandere2x\\src\\workspace\\yn_moving.mkv']
get frame count slow
```

Right?

**Comment by akai-katto** [comment]
Where are you located? Are you interested in hoping into a discord call and sorting this out?

**Comment by eoiz** [comment]
Hi! Sorry for the late response but I've been really busy with work.I live in Italy, I don't speak fluently english so that might be difficult but thanks for the offer :)Chat should do it.Il giorno 13 set 2022, alle ore 6:43, Tyler ***@***.***> ha scritto:Where are you located? Are you interested in hoping into a discord call and sorting this out?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by DCGYT** [comment]
> '-v', 'panic', '-print_format', 'json', '-show_format', '-show_streams', '-i',

![image](https://user-images.githubusercontent.com/98280315/192907206-1ea7efe1-3295-48fa-84b6-f68583174aa4.png)
Sorry for the late, but I don't see the message (last main you sent).

**Comment by akai-katto** [comment]
Sorry for the late response, I was on a business trip. 

It seems I uploaded the wrong binary. I'll update it tonight.

**Comment by akai-katto** [comment]
It seems I removed the print statement on accident (not on accident, it's just been a while). 

Does the binary I sent you work? it works by manually counting the frames in a video, hoping it works compared to the other solutions. 

Again, sorry for the delayed response, was on a business trip.

**Comment by DCGYT** [comment]
@akai-katto Using the normal 3.6 (downloaded from the Tags): I obtained the next results:
10792 frames - CPU 99% 2.3Ghz [First attempt]
16942 frames - CPU 99% 2.3Ghz
17485 frames - CPU 99% 2.3Ghz
28391 frames - TurboBoost 4Ghz
25790 frames - Turbo Boost 4Ghz
15700 frames - Without connected CPU 99%
12872 frames - Battery Saver CPU 99%
12192 frames - Battery Saver CPU 99%

**Comment by akai-katto** [comment]
Any update on the custom binary?

**Comment by akai-katto** [comment]
https://github.com/akai-katto/dandere2x/releases/tag/3.7

The long video error should be fixed in 3.7.2

**Comment by DCGYT** [comment]
> https://github.com/akai-katto/dandere2x/releases/tag/3.7
> 
> The long video error should be fixed in 3.7.2

Thank you very much, it also worked for me.

I am working on making a definitive list with information about the bugs, I will start testing with versions from 1.3 to 3.7 to see how the bugs have been fixed and to have information about when new ones appeared.
![image](https://user-images.githubusercontent.com/98280315/199132111-545ff6e1-a931-4635-9185-151d99d8d5e8.png)


P.S.: could you please close this post, I lost track of where I wrote it.
![image](https://user-images.githubusercontent.com/98280315/199132340-51a59d64-1989-4f34-8347-3911ab20bec6.png)
